### PR TITLE
Added unittests for FifoArray.

### PR DESCRIPTION
**Pull request by lewie-donckers**
Added some unittests for `FifoArray<>` to get the ball rolling.

If I understand the catkin build environment correctly, these unittests will be built and run when the other tests are run. For example when you use:
```sh
catkin run_tests
```` 
or
```sh
rostest path_tracking_pid test_path_tracking_pid.test
```

You can also build them separately:
```sh
catkin build -i --make-args unittests
```

And run them separately:
```sh
rosrun path_tracking_pid unittests
```

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -106,4 +106,6 @@ install(
 
 if(CATKIN_ENABLE_TESTING)
   add_rostest(test/test_path_tracking_pid.test ARGS rviz:=false reconfigure:=false)
+  catkin_add_gtest(unittests unittest/test_main.cpp unittest/test_fifo_array.cpp)
+  target_link_libraries(unittests ${catkin_LIBRARIES})
```
Comment (Rayman) [inline CMakeLists.txt:110]: I would keep all tests in the `test` folder to keep to the standard package structure. So for example `test/unittests/...`

**Review by Rayman — CHANGES_REQUESTED** [review]
Nice, looks good

**Comment by Rayman** [comment]
- `catkin run_tests --this --no-deps` builds & runs the tests
- `catkin_test_results` gets the results of previously ran tests
- `catkin test  --this --no-deps` runs tests and gets the results

File: CMakeLists.txt (hunk 1 above)
Comment (lewie-donckers) [reply CMakeLists.txt:110]: @Rayman fixed!

**Review by Timple — APPROVED** [review]
Awesome, you're getting the hang of the ROS compilation quirks/features :slightly_smiling_face:.

unittests where on the todo list for a long while now, thanks!

**Review by Rayman — APPROVED** [review]

**Pull request merged**